test/e2e/allocation.new.test.js

```diff
@@ -0,0 +1,40 @@
+import { pmuUser } from './_roles'
+import { newAllocation } from './_routes'
+import { allocationJourney } from './pages/'
+
+fixture('New PMU allocation').beforeEach(async t => {
+  await t.useRole(pmuUser).navigateTo(newAllocation)
+})
+test('Create allocation and verify the result', async t => {
+  await allocationJourney.createAllocation()
+  const currentUrl = await allocationJourney.getCurrentUrl()
```

Comment:
I have a feeling this is the kind of thing that caused some flakiness with tests previously.

I think it's safer to call `getCurrentUrl()` within the expect call:

```javascript
await t
    .expect(allocationJourney.getCurrentUrl())
    .match(...)
```

Reply:
This is not a match as in, expect something to match. It's match as in, extract a piece of a string. How that would work if I put it in an expect block?

Reply:
Btw, so far this particular step has caused no flakiness whatsoever. The only thing that has created problems is the interaction with elements off the viewport.

Reply:
What's the goal? To navigate to the actual allocation using an ID?

It feels like it would be nicer and cleaner if `allocationJourney.createAllocation()` returned you the allocation, along with the ID. But I get that's its not as easy as the other fixtures because it's not hitting the API and therefore returning the ID.

How about placing the ID somewhere in the DOM? We are [doing something](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/commit/23eb7e810f18c5117dc1923c10dfcd4480f23713) similar to know which current location a user has selected and it would be a nicer way to get that information than having to get it from the URL via a regex. I think the `getCurrentUrl` hasn't felt robust enough so the more we can avoid using it, I think, the better.

You could also add it to the template using a `data-` attribute to something like the confirmation banner. That way, in `allocationJourney.createAllocation()` you can retrieve it from there and return it as part of that method.

Reply:
After that chat on slack I think just moving this logic to the `createAllocation` would be enough here and would tidy this up a little.

So that method then returns the ID and we have it contained in one place.